crates/circuit/src/circuit_drawer.rs

```diff
@@ -0,0 +1,1103 @@
+        }
+    }
+
+    fn pad_string(s: &mut String, pad_char: char, width: usize){
+        let current_width = s.chars().count();
+        if current_width < width{
+            let pad_size = width - current_width;
+            let left_pad = pad_size / 2;
+            let right_pad = pad_size - left_pad;
+            let left_pad_str = pad_char.to_string().repeat(left_pad);
+            let right_pad_str = pad_char.to_string().repeat(right_pad);
+            let new_str = format!("{}{}{}", left_pad_str, s, right_pad_str);
+            *s = new_str;
+        }
+    }
+
+    fn pad_wire(&mut self, mid_char: char, width: usize){
+        let current_width = self.width();
+        if current_width < width{
+            // let pad_size = width - current_width;
+            // let left_pad = pad_size / 2 - 1;
+            // let right_pad = pad_size - left_pad - 1;
+            // let left_pad_str = mid_char.to_string().repeat(left_pad);
+            // let right_pad_str = mid_char.to_string().repeat(right_pad);
+            // self.top = format!("{}{}{}", " ".to_string().repeat(left_pad), self.top, " ".to_string().repeat(right_pad));
+            // self.mid = format!("{}{}{}", left_pad_str, self.mid, right_pad_str);
+            Self::pad_string(&mut self.top, ' ', width);
+            Self::pad_string(&mut self.mid, mid_char, width);
+            Self::pad_string(&mut self.bot, ' ', width);
+        }
+        //println!("layer width:{}, object width:{} : \n{}\n{}\n{}", width, current_width, self.top, self.mid, self.bot);
+    }
+}
+
+struct CircuitRep{
+    wires: Vec<Wire>,
+}
+
+impl CircuitRep{
+    fn print(&self){
+        for wire in &self.wires{
+            println!("{}\n{}\n{}", wire.top, wire.mid, wire.bot);
+        }
+    }
+
+    fn add_wire(&mut self, wire: &Wire, ind: usize){
+        self.wires[ind].top.push_str(&wire.top);
+        self.wires[ind].mid.push_str(&wire.mid);
+        self.wires[ind].bot.push_str(&wire.bot);
+        // self.wires[ind].top.push_str(&format!("{}{}",&wire.top,"$"));
+        // self.wires[ind].mid.push_str(&format!("{}{}",&wire.mid,"$"));
+        // self.wires[ind].bot.push_str(&format!("{}{}",&wire.bot,"$"));
+    }
+}
+
+impl Index<usize> for CircuitRep{
+    type Output = Wire;
+
+    fn index(&self, index: usize) -> &Self::Output {
+        &self.wires[index]
+    }
+}
+
+pub const Q_WIRE: char = '─';
+pub const C_WIRE: char = '═';
+pub const TOP_CON: char = '┴';
+pub const BOT_CON: char = '┬';
+pub const Q_LEFT_CON: char = '┤';
+pub const Q_RIGHT_CON: char = '├';
+pub const CL_LEFT_CON: char = '╡';
+pub const CL_RIGHT_CON: char = '╞';
+pub const TOP_LEFT_BOX: char = '┌';
+pub const TOP_RIGHT_BOX: char = '┐';
+pub const BOT_LEFT_BOX: char = '└';
+pub const BOT_RIGHT_BOX: char = '┘';
+pub const BARRIER: char = '░';
+pub const BULLET:char = '■';
+pub const CONNECTING_WIRE:char = '│';
+pub const CROSSED_WIRE:char = '┼';
+pub const CL_CROSSED_WIRE:char = '╪';
+
+struct TextDrawer;
```

Comment:
The struct is used merely as a namespace, as all it's functions are associated functions. 
As suggested in another comment - why don't we move `Wire` (to be renamed `LayerWire`) data into the struct? I assume it'll need more state data. So beside the constructor probably all the other functions should be methods.